gluster-setup: retry in case mounting fails

nixpanic opened the pull request:
Currently gluster-setup.sh aborts when 'mount' returns an error. In
certain cases LVM/LVs are not available yet, and mounting the devices is
not possible. The script already tries to account for that, but a
mistake prevents the retry from being reached.

With this change the gluster-setup.sh script exits only when mounting
succeeds, and will do the retry (including 'vgscan --mknods') when a
failure occured.

Signed-off-by: Niels de Vos <ndevos@redhat.com>

phlogistonjohn commented:
Can we split the clean up bits and the fixes into different commits so the essential parts of the fix are more prominent?

humblec commented:
@nixpanic I do agree with @phlogistonjohn , lets split the commits to cleanup and other parts.

nixpanic commented:
On Tue, Dec 18, 2018 at 09:57:51PM +0000, John Mulligan wrote:
> Can we split the clean up bits and the fixes into different commits so the essential parts of the fix are more prominent?

Done. Split in (1) if-mount-success and (2) do-not-log-empty-lines.

ansiwen commented on CentOS/gluster-setup.sh:
```diff
@@ -39,14 +39,14 @@ main () {
-            echo "" > $GLUSTERFS_LOG_CONT_DIR/failed_bricks
-            echo "" > $GLUSTERFS_LOG_CONT_DIR/lvscan
-            echo "" > $GLUSTERFS_LOG_CONT_DIR/mountfstab
+        true > $GLUSTERFS_LOG_CONT_DIR/brickattr
```
The old comments seem to have disappeared (normally outdated comments should still be referenced in the PR, no?), so here again:
I actually like `:> file` or even `:>| file` as proposed here: https://unix.stackexchange.com/questions/130971/difference-between-cat-and-to-zero-out-a-file. Looks like a big nose unhappy smiley. :-)

ansiwen commented on CentOS/gluster-setup.sh:
```diff
@@ -58,12 +58,14 @@ main () {
         then
-              echo "mount command exited with code ${sts}" >> $GLUSTERFS_LOG_CONT_DIR/mountfstab
-              exit 1
+              echo "Mount command Successful" >> $GLUSTERFS_LOG_CONT_DIR/mountfstab
```
typo: ~M~mount command ~S~successful

ansiwen commented on Fedora/gluster-setup.sh:
```diff
@@ -56,12 +56,14 @@ main () {
         then
-              echo "mount command exited with code ${sts}" >> $GLUSTERFS_LOG_CONT_DIR/mountfstab
-              exit 1
+              echo "Mount command Successful" >> $GLUSTERFS_LOG_CONT_DIR/mountfstab
```
typo: ~M~mount command ~S~successful

ansiwen commented on CentOS/gluster-setup.sh:
```diff
@@ -39,14 +39,14 @@ main () {
         mkdir $GLUSTERFS_LOG_CONT_DIR
-        echo "" > $GLUSTERFS_LOG_CONT_DIR/brickattr
-        echo "" > $GLUSTERFS_LOG_CONT_DIR/failed_bricks
+        true > $GLUSTERFS_LOG_CONT_DIR/brickattr
```
While at it, I would move brickattr and failed_bricks out of the if/else clauses
